src/netex_idf/stops.rs

```diff
@@ -83,8 +110,109 @@ fn load_stop_areas(elem: &Element) -> Result<CollectionWithId<StopArea>> {
             );
         }
     }
+}
 
-    Ok(stop_areas)
+fn load_coords(quay: &Element) -> Result<(f64, f64)> {
+    let gml_pos = quay
+        .try_only_child("Centroid")?
+        .try_only_child("Location")?
+        .try_only_child("pos")?
+        .text()
+        .trim()
+        .to_string();
+
+    let coords: Vec<&str> = gml_pos.split_whitespace().collect();
+    if coords.len() != 2 {
+        bail!("longitude and latitude not found");
+    }
+
+    Ok((coords[0].parse()?, coords[1].parse()?))
+}
+
+fn load_stop_points<'a>(
+    quays: impl Iterator<Item = &'a &'a Element>,
+    stop_areas: &mut CollectionWithId<StopArea>,
+    map_quay_lda: &mut HashMap<String, String>,
+) -> Result<CollectionWithId<StopPoint>> {
+    let mut stop_points = CollectionWithId::default();
+    // add stop points
+    let from = "EPSG:2154";
+    let to = "+proj=longlat +datum=WGS84 +no_defs";
+    let proj = Proj::new_known_crs(&from, &to, None)
+        .ok_or_else(|| format_err!("Proj cannot build a converter from '{}' to '{}'", from, to))?;
+
+    for quay in quays {
+        let id: String = quay.try_attribute("id")?;
+        let coords = skip_fail!(load_coords(quay).map_err(|e| format_err!(
+            "unable to parse coordinates of quay {}: {}",
+            id,
+            e
+        )));
+
+        let mut stop_point = StopPoint {
+            id: quay.try_attribute("id")?,
+            name: quay.try_only_child("Name")?.text().trim().to_string(),
+            visible: true,
+            coord: proj.convert((coords.0, coords.1).into()).map(Coord::from)?,
+            stop_area_id: "default_id".to_string(),
+            timezone: Some(EUROPE_PARIS_TIMEZONE.to_string()),
+            stop_type: StopType::Point,
+            ..Default::default()
+        };
+
+        let stop_point = if let Some(stop_area_id) =
+            map_quay_lda.get(&id).and_then(|stop_area_id| {
+                stop_areas
+                    .get(&stop_area_id)
+                    .map(|_| stop_area_id.to_string())
+            }) {
+            StopPoint {
+                stop_area_id,
+                ..stop_point
+            }
+        } else {
+            let stop_area = StopArea::from(stop_point.clone());
+            stop_point.stop_area_id = stop_area.id.clone();
+            stop_areas.push(stop_area)?;
+            stop_point
+        };
+
+        stop_points.push(stop_point)?;
+    }
+
+    Ok(stop_points)
+}
+
+fn load_stops(elem: &Element) -> Result<(CollectionWithId<StopArea>, CollectionWithId<StopPoint>)> {
+    let member_children: Vec<_> = elem
+        .try_only_child("dataObjects")?
+        .try_only_child("CompositeFrame")?
+        .try_only_child("frames")?
+        .children()
+        .flat_map(|e| e.children())
+        .filter(|e| e.name() == "members")
+        .flat_map(|e| e.children())
+        .collect();
```

Comment:
Does it need to be collected? Maybe you could keep the iterator.
```
let mut members = elem
        .try_only_child("dataObjects")?
        .try_only_child("CompositeFrame")?
        .try_only_child("frames")?
        .children()
        .flat_map(|e| e.children())
        .filter(|e| e.name() == "members")
        .flat_map(|e| e.children())
```
Then you could send an iterator to `load_stop_areas` and `load_stop_points`.
```
let mut members_stops_places = members.clone().filter(|e| e.name() == "StopPlace");
let mut stop_areas = load_stop_areas(members_stop_places, &mut map_lda_zde, &mut map_quay_lda)?;
...
let mut members_quays = members.clone().filter(|e| e.name() == "Quay");
let stop_points = load_stop_points(member_quays, &mut stop_areas, &mut map_quay_lda)?;
```

Reply:
I don't see the difference between collecting and not collecting + clone

Reply:
An iterator is probably much lighter in memory space than a `Vec`. And I did use the `.clone()` twice but technically, only one is needed (on the second use, you can just do `members.filter(...)`.

Reply:
it seems I can't do that.
```
200 |     let members_stops_places = member_children
    |                                --------------- value moved here
...
207 |     let members_quays = member_children.cloned().filter(|e| e.name() == "Quay");
    |                         ^^^^^^^^^^^^^^^ value used here after move
```

Reply:
I was talking about `.clone()` which would clone the iterator, not `cloned()` that would clone the items inside the iterator.

Reply:
ok I thought it was a mistake because clone is not implemented by FlatMap

Reply:
Hmm, it is implemented from what I can see here
https://doc.rust-lang.org/std/iter/struct.FlatMap.html#impl-Clone

Reply:
you are right but `minidom::element::Children` should also  implement `Clone` but it does not.